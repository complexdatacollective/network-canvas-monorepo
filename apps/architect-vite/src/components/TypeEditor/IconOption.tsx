import Icon from "@codaco/legacy-ui/components/Icon";
import Radio from "../Form/Fields/Radio";

type IconOptionProps = {
	label: string;
} & React.ComponentProps<typeof Radio>;

const IconOption = (props: IconOptionProps) => (
	<Radio
		// eslint-disable-next-line react/jsx-props-no-spreading
		{...props}
		className="type-editor-icon-option"
		// eslint-disable-next-line react/destructuring-assignment
		label={<Icon name={props.label} />}
	/>
);

export default IconOption;
