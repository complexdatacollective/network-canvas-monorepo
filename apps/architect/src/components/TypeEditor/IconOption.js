import React from 'react';
import PropTypes from 'prop-types';
import Icon from '@codaco/ui/lib/components/Icon';
import Radio from '../Form/Fields/Radio';

const IconOption = (props) => (
  <Radio
    // eslint-disable-next-line react/jsx-props-no-spreading
    {...props}
    className="type-editor-icon-option"
    // eslint-disable-next-line react/destructuring-assignment
    label={<Icon name={props.label} />}
  />
);

IconOption.propTypes = {
  label: PropTypes.string.isRequired,
};

export default IconOption;
