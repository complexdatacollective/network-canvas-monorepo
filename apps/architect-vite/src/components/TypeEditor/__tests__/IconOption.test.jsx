/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import IconOption from '../IconOption';

const mockProps = {
  label: 'add-a-person',
  input: {},
};

describe('<IconOption />', () => {
  it('can render', () => {
    const subject = shallow(<IconOption {...mockProps} />);
    expect(subject).toMatchSnapshot();
  });
});
