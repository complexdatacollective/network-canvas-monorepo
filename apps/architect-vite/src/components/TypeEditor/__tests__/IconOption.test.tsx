import { describe, it, expect } from "vitest";

import { render } from "@testing-library/react";
import IconOption from "../IconOption";

const mockProps = {
	label: "add-a-person",
	input: {},
};

describe("<IconOption />", () => {
	it("can render", () => {
		const { container } = render(<IconOption {...mockProps} />);
		expect(container.firstChild).toBeInTheDocument();
	});
});
