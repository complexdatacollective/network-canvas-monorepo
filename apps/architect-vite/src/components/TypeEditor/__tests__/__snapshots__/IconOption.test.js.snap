// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconOption /> can render 1`] = `
<Radio
  className="type-editor-icon-option"
  disabled={false}
  fieldLabel={null}
  input={Object {}}
  label={
    <Memo(Icon)
      name="add-a-person"
    />
  }
/>
`;
